pkg/gcmanager/scanAll_IPPool.go

```diff
@@ -120,142 +118,220 @@ func (s *SpiderGC) executeScanAll(ctx context.Context) {
 					zap.String("podUID", poolIPAllocation.PodUID),
 				)
 
+				flagGCIPPoolIP := false
+				flagGCEndpoint := false
+				flagPodStatusShouldGCIP := false
+				flagTracePodEntry := false
+				flagStaticIPPod := false
 				podYaml, err := s.podMgr.GetPodByName(ctx, podNS, podName, constant.UseCache)
+				endpoint, endpointErr := s.wepMgr.GetEndpointByName(ctx, podNS, podName, constant.UseCache)
 				if err != nil {
 					// case: The pod in IPPool's ip-allocationDetail is not exist in k8s
 					if apierrors.IsNotFound(err) {
-						wrappedLog := scanAllLogger.With(zap.String("gc-reason", "pod not found in k8s but still exists in IPPool allocation"))
-						endpoint, err := s.wepMgr.GetEndpointByName(ctx, podNS, podName, constant.UseCache)
-						if nil != err {
-							// just continue if we meet other errors
-							if !apierrors.IsNotFound(err) {
-								wrappedLog.Sugar().Errorf("failed to get SpiderEndpoint: %v", err)
+						if endpointErr != nil {
+							if apierrors.IsNotFound(endpointErr) {
+								scanAllLogger.Sugar().Infof("pod %s/%s and endpoint %s/%s not found, only recycle IPs in IPPool'", podNS, podName, podNS, podName)
+								flagGCIPPoolIP = true
+								flagGCEndpoint = false
+								goto GCIP
+							} else {
+								scanAllLogger.Sugar().Errorf("pod not found and failed to get endpoint %s/%s, error '%v'", podNS, podName, err)
 								continue
 							}
 						} else {
-							if s.gcConfig.EnableStatefulSet && endpoint.Status.OwnerControllerType == constant.KindStatefulSet {
-								isValidStsPod, err := s.stsMgr.IsValidStatefulSetPod(ctx, podNS, podName, constant.KindStatefulSet)
-								if nil != err {
-									scanAllLogger.Sugar().Errorf("failed to check StatefulSet pod IP '%s' should be cleaned or not, error: %v", poolIP, err)
-									continue
-								}
-								if isValidStsPod {
-									scanAllLogger.Sugar().Warnf("no need to release IP '%s' for StatefulSet pod ", poolIP)
-									continue
-								}
+							vaildPod, err := s.isValidStaticPod(ctx, scanAllLogger, podNS, podName, poolIP, endpoint.Status.OwnerControllerType)
+							if err != nil {
+								scanAllLogger.Sugar().Debugf("pod %v/%v not found, error: %v", podNS, podName, err)
```

Comment:
+1

Reply:
scanAllLogger.Sugar().Debugf("pod %s/%s does not exist, but the pod is a valid static pod, ignore handle IP %s and endpoint %s/%s", podNS, podName, poolIP, podNS, podName)